pageserver/src/basebackup.rs

```diff
@@ -255,6 +259,22 @@ where
     async fn send_tarball(mut self) -> Result<(), BasebackupError> {
         // TODO include checksum
 
+        // Detect if we are creating the basebackup exactly at a shutdown checkpoint.
+        let normal_shutdown =
+            if let Ok(checkpoint_bytes) = self.timeline.get_checkpoint(self.lsn, self.ctx).await {
+                let checkpoint =
+                    CheckPoint::decode(&checkpoint_bytes).context("deserialize checkpoint")?;
+                let checkpoint_end_lsn = calculate_walrecord_end_lsn(
+                    Lsn(checkpoint.redo),
+                    XLOG_SIZE_OF_XLOG_RECORD
+                        + SIZE_OF_XLOG_RECORD_DATA_HEADER_SHORT
+                        + SIZEOF_CHECKPOINT,
+                );
+                checkpoint_end_lsn == self.lsn
```

Comment:
Is this correct when the checkpoint record ends exactly at page boundary? Does `self.lsn` point to the beginning of the page or just after the page header in that case?

Reply:
This code is removed now